tests/chunk.cpp

```diff
@@ -520,6 +520,21 @@ TEST(Chunk2DTest, SliceCoordY)
     }
 }
 
+TEST(Chunk2DTest, SliceCoordXOutOfBounds)
+{
+#if !defined(NDEBUG) // The assertion is only checked if NDEBUG isn't defined
+    ChunkXY<double> chunk(dom_x_y);
+
+    char const* const death_msg
+            = R"rgx([Aa]ssert.*select<QueryDDims...>\(this->m_domain\).contains\(slice_spec\))rgx";
+
+    EXPECT_DEATH(chunk[dom_x.front() - 1], death_msg);
+    EXPECT_DEATH(chunk[dom_x.back() + 1], death_msg);
```

Comment:
In case of array out of bounds, is there a way to identify which dimension involves out of bounds access?